test/constraints-validation.test.ts

```diff
@@ -0,0 +1,190 @@
+import { z, ZodError } from 'zod';
+import { KubernetesVersion } from 'aws-cdk-lib/aws-eks/lib';
+
+const app = new cdk.App();
+
+const addOns: Array<blueprints.ClusterAddOn> = [
+  new blueprints.addons.MetricsServerAddOn(),
+  new blueprints.addons.ClusterAutoScalerAddOn(),
+  new blueprints.addons.VpcCniAddOn(),
+];
+
+const tooBigNumber5000 = new z.ZodError([{
+  code: "too_big",
+  maximum: 5000,
+  type: "number",
+  inclusive: true,
+  path: [],
+  message: ""
+}]);
+
+const tooBigString63 = new z.ZodError([{
+  code: "too_big",
+  maximum: 63,
+  type: "string",
+  inclusive: true,
+  path: [],
+  message: ""
+}]);
+
+const tooSmallString1 = new z.ZodError([{
+  code: "too_small",
+  minimum: 1,
+  type: "string",
+  inclusive: true,
+  path: [],
+  message: ""
+}]);
+
+const tooSmallNumber0 = new z.ZodError([{
+  code: "too_small",
+  minimum: 0,
+  type: "number",
+  inclusive: true,
+  path: [],
+  message: ""
+}]);
+
+const notUrlError = new z.ZodError([{
+  validation: "url",
+  code: "invalid_string",
+  message: "",
+  path: []
+}]);
+
+const longName = 'eks-blueprint-with-an-extra-kind-of-very-long-name-that-is-definitely-not-going-to-work';
+
+const blueprintBase = blueprints.EksBlueprint.builder().addOns(...addOns);
+
+type Executable = () => void;
+
+type DataError = [Executable, ZodError];
+
+function createAutoScalingGroup(id: string, minSize?: number, maxSize?: number, desiredSize?: number) {
+  return new blueprints.GenericClusterProvider({
+    version: KubernetesVersion.V1_21,
+    autoscalingNodeGroups: [
+      {
+        id: id,
+        minSize: minSize,
+        maxSize: maxSize,
+        desiredSize: desiredSize
+      }]
+  });
+}
+
+function createManyAutoScalingGroup(length: number) {
+  return new blueprints.GenericClusterProvider({
+    version: KubernetesVersion.V1_21,
+    autoscalingNodeGroups: loop(new Array<AutoscalingNodeGroup>(length).fill({ id: "" }))
+  });
+}
+
+function loop(array: Array<AutoscalingNodeGroup>): AutoscalingNodeGroup[] {
+  for (let i = 0; i < array.length; i++) {
+    array[i] = { id: "" + i };
+  }
+  return array;
+}
+
+function singleErrorInArray(id: string, errorNumberVariable?: number) {
+  return new blueprints.GenericClusterProvider({
+    version: KubernetesVersion.V1_21,
+    autoscalingNodeGroups: [
+      {
+        id: id + 1
+      },
+      {
+        id: id + 2
+      },
+      {
+        id: id + 3
+      },
+      {
+        id: id + 4,
+        minSize: errorNumberVariable
+      }
+    ]
+  });
+}
+
+function createFargateProfile(fargateProfileName: string) {
+  return new blueprints.GenericClusterProvider({
+    version: KubernetesVersion.V1_21,
+    fargateProfiles: {
+      "fp1": {
+        fargateProfileName: fargateProfileName,
+        selectors: [{ namespace: "serverless1" }]
+      }
+    }
+  });
+}
+function getConstraintsDataSet(): DataError[] {
+
+  let result: [Executable, ZodError][] = [];
```

Comment:
4 spaces is the indentation that we use

Reply:
I'm sorry what do you mean? Like 4 spaces before let? I just used the Format Document feature and it does it like this. Or do you mean in the space between function and let perhaps?

Reply:
Scratch that, it was an aberration in the GitHub view - your tab size looked like it was "2". We adopted tab size "4" and spaces instead of tabs.